Calibration/EcalAlCaRecoProducers/plugins/AlCaElectronTracksReducer.cc

```diff
@@ -51,11 +51,7 @@ void AlCaElectronTracksReducer::produce (edm::Event& iEvent,
 
   for (eleIt=electronCollection->begin(); eleIt!=electronCollection->end(); eleIt++) {
     // barrel
-#ifndef CMSSW42X
-    TrackRef track = (eleIt-> closestTrack() ) ;
-#else
     TrackRef track = (eleIt-> closestCtfTrackRef());
```

Comment:
Is the change in the logic here intentional? L54 had an `#ifndef` statement, not an `#ifdef`.

Reply:
there is no logic change. 

closestTrack() no longer exists and was removed as it is broken for pat::Electrons. For reco::GsfElectrons, its the same as closestCtfTrackRef().

Reply:
The purpose of the change here is due to removal of ```closestTrack``` method, which was always equivalent for reco::GsfElectrons but was also always broken for pat::Electrons.

```closestTrack``` was introduced for PAT purposes in 5_2_0 but was apparently never made useful for PAT (and was even buggy for the pat::Electrons).
```closestCtrTrackRef``` existed since CMSSW_3_1_0.